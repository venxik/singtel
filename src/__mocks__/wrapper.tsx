import { Provider } from 'react-redux';
import { AppStore, RootState, setupStore } from '../redux';
import { PreloadedState } from '@reduxjs/toolkit';
import { PropsWithChildren } from 'react';
import { render } from '@testing-library/react-native';

interface ExtendedRenderOptions {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function AllTheProviders({ children }: { children: JSX.Element }) {
  return <Provider store={setupStore()}>{children}</Provider>;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
