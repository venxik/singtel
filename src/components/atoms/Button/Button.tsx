import {
  TouchableOpacityProps as DefaultTouchableOpacityProps,
  TouchableOpacity,
} from 'react-native';
import { ThemeProps, useThemeColor } from '../Themed/Themed';

export type TouchableOpacityProps = ThemeProps & DefaultTouchableOpacityProps;

const Button = (props: TouchableOpacityProps) => {
  const { style, lightColor, darkColor, disabled, ...otherProps } = props;
  const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, 'buttonBackground');
  const backgroundColorDisabled = useThemeColor(
    { light: lightColor, dark: darkColor },
    'buttonDisabledBackground'
  );

  return (
    <TouchableOpacity
      style={[
        {
          backgroundColor: disabled ? backgroundColorDisabled : backgroundColor,
          borderRadius: 20,
          paddingVertical: '5%',
          paddingHorizontal: '15%',
          alignItems: 'center',
        },
        style,
      ]}
      disabled={disabled}
      {...otherProps}>
      {props.children}
    </TouchableOpacity>
  );
};

export default Button;
