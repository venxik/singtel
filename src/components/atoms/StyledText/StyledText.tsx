import { Text, TextProps } from '../Themed/Themed';

const ComicText = (props: TextProps) => {
  return <Text {...props} style={[props.style, { fontFamily: 'comic' }]} />;
};

export default ComicText;
