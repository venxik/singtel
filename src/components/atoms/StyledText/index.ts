import ComicText from './StyledText';

export { ComicText };
