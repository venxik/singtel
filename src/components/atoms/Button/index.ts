import Button, { TouchableOpacityProps } from './Button';

export { Button, TouchableOpacityProps };
