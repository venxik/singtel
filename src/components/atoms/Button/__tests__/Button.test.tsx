import React from 'react';
import { renderWithProviders } from '../../../../__mocks__';
import Button from '../Button';
import { ComicText } from '../../StyledText';
import { fireEvent, screen } from '@testing-library/react-native';

describe('Button', () => {
  it(`match snapshot`, () => {
    renderWithProviders(
      <Button>
        <ComicText>Test</ComicText>
      </Button>
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it(`can be pressed and functional`, () => {
    let number = 1;
    renderWithProviders(
      <Button testID="button" onPress={() => (number += 1)}>
        <ComicText>Test</ComicText>
      </Button>
    );
    fireEvent.press(screen.getByTestId('button'));
    expect(number).toBe(2);
  });
});
