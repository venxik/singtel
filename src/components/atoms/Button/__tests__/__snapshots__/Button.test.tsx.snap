// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#686fcc",
      "borderRadius": 20,
      "opacity": 1,
      "paddingHorizontal": "15%",
      "paddingVertical": "5%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        [
          undefined,
          {
            "fontFamily": "comic",
          },
        ],
      ]
    }
  >
    Test
  </Text>
</View>
`;
